addons/spreadsheet/static/src/chart/odoo_menu/figure_component.js

```diff
@@ -3,18 +3,29 @@
 import { patch } from "@web/core/utils/patch";
 import * as spreadsheet from "@odoo/o-spreadsheet";
 import { useService } from "@web/core/utils/hooks";
+import { _t } from "@web/core/l10n/translation";
 
 patch(spreadsheet.components.FigureComponent.prototype, {
     setup() {
         super.setup();
         this.menuService = useService("menu");
         this.actionService = useService("action");
+        this.notificationService = useService("notification");
     },
     async navigateToOdooMenu() {
         const menu = this.env.model.getters.getChartOdooMenu(this.props.figure.id);
         if (!menu) {
             throw new Error(`Cannot find any menu associated with the chart`);
         }
+        if (!menu.actionID) {
+            this.notificationService.add(
+                _t(
+                    "The menu linked to this chart doesn't have an corresponding action. Please link the chart to another menu."
+                ),
```

Comment:
Aren't those menu already filtered by the domain ? (`getDomain` in `ir_menu_selector.js`)
It's better to avoid ending in this situation in the first place (and avoid work for translators)

Reply:
That's filtered when creating the menu, but the codebase is moving. The dashboard "sales" was fine at its creation, but then became buggy because the menu changed in master. For dashboards it shouldn't happen anymroe because of the python tests, but a random spreadsheet created by an user can break if he upgrades odoo or change himself some code.